src/MVVM/PropertyChanged/BindingBase/BindingBase.propertyChanged.cs

```diff
@@ -200,34 +215,81 @@ private IEnumerable<PropertyInfo> GetCommandProperties()
                         propertyInfo.PropertyType is ICommand
                         || typeof(ICommand).IsAssignableFrom(propertyInfo.PropertyType));
 
+
             return commandProperties;
         }
 
-        private void UpdateCommandsCanExecute()
+        private void UpdateCommandsCanExecute(
+            CommandRelevantPropertyWrapper commandRelevance)
         {
-            foreach (var property in GetCommandProperties())
+            if (commandRelevance == null)
             {
-                if (!(property.GetValue(this) is ICommand command))
-                    return;
+                return;
+            }
 
-                var eventDelegate = (MulticastDelegate)command
-                    .GetType()
-                    .GetField(
-                        nameof(command.CanExecuteChanged),
-                        BindingFlags.Instance | BindingFlags.NonPublic)
-                    .GetValue(command);
 
-                if (eventDelegate == null)
-                    return;
+            if (!string.IsNullOrWhiteSpace(
+                commandRelevance.CommandName))
+            {
+                var commandProperty = GetCommandProperties()
+                    .FirstOrDefault(
+                        property => property.Name == commandRelevance.CommandName);
 
-                foreach (var eventHandler in eventDelegate.GetInvocationList())
+
+                if (commandProperty?.GetValue(this) is ICommand command)
                 {
-                    eventHandler.Method
-                        .Invoke(
-                            eventHandler.Target,
-                            new object[] { command, EventArgs.Empty });
+                    InvokeCommandCanExecuteChanged(
+                        command,
+                        commandProperty.Name);
+
+                    return;
                 }
             }
+
+
+            foreach (var property in GetCommandProperties())
+            {
+                if (!(property.GetValue(this) is ICommand command))
+                    continue;
+
+
+                InvokeCommandCanExecuteChanged(
+                    command,
+                    property.Name);
+            }
+        }
+
+        private void InvokeCommandCanExecuteChanged(
+            ICommand command,
+            string propertyName)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'propertyName'.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5640474)